src/main/resources/Office365/Groovy.xml

```diff
@@ -335,13 +334,13 @@ public class AzureAuth {
      } else {
        debug("Step 2b: Got access token succesfully")
        storeAuthentication(request, authResult);
-       if (request.state &amp;&amp; request.state!="") {
-         def url = storedStates.get(request.state);
-         if (url!=null) {
-            debug("Redirecting to final destination after authorization: ${url}")
-            response.sendRedirect(url)
-            return true;
-         }
+       def session = request.getSession();
+       def url = session.getAttribute(AZURE_SESSION_REDIRECT);
+       if(url &amp;&amp; url.length()&gt;=0) {
```

Comment:
The old code style was a good one. Please make sure you follow the same. Indentation, blank space between `if` and `{`, space between terms that are compared `url.length() >= 0` instead of `url.length>=0`. The more code is written the bigger is the gain of these small improvements.

Reply:
Fixed in https://github.com/xwikisas/application-office365/pull/11/commits/73ead5a9b328d8ac9d7dd47a720688076251f346 . thanks to check.